package/gluon-mesh-vpn-fastd/src/respondd.c

```diff
@@ -0,0 +1,301 @@
+/*
+  Copyright (c) 2016, Matthias Schiffer <mschiffer@universe-factory.net>
+  All rights reserved.
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are met:
+
+    1. Redistributions of source code must retain the above copyright notice,
+       this list of conditions and the following disclaimer.
+    2. Redistributions in binary form must reproduce the above copyright notice,
+       this list of conditions and the following disclaimer in the documentation
+       and/or other materials provided with the distribution.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+  FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+
+#include <respondd.h>
+
+#include <json-c/json.h>
+#include <libgluonutil.h>
+#include <uci.h>
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+#include <sys/socket.h>
+#include <sys/un.h>
+
+
+static struct json_object * get_peer_groups(struct json_object *groups, struct json_object *peers);
+
+static struct json_object * get_fastd_version(void) {
+        FILE *f = popen("exec fastd -v", "r");
+        if (!f)
+                return NULL;
+
+        char *line = NULL;
+        size_t len = 0;
+
+        ssize_t r = getline(&line, &len, f);
+
+        pclose(f);
+
+        if (r >= 0) {
+                len = strlen(line);
+
+                if (len && line[len-1] == '\n')
+                        line[len-1] = 0;
+        }
+        else {
+                free(line);
+                line = NULL;
+        }
+
+        struct json_object *ret = gluonutil_wrap_string(line + 6);
+        free(line);
+        return ret;
+}
+
+static struct json_object * get_fastd(void) {
+        bool enabled = false;
+
+        struct uci_context *ctx = uci_alloc_context();
```

Comment:
I'd propose packaging `uci_alloc_context()`, the setting of the strict flag and `uci_load()` into a libgluonutil function. It occurs four times in this PR, and especially the setting of the strict flag could easily be forgotten otherwise.